File: backend/src/main/java/com/interviewme/model/Company.java
package com.interviewme.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Filter;

import java.time.OffsetDateTime;

@Entity
@Table(name = "company")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Filter(name = "tenantFilter", condition = "tenant_id = :tenantId")
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(length = 20, unique = true)
    private String cnpj;

    @Column(length = 500)
    private String website;

    @Column(length = 100)
    private String sector;

    @Column(length = 20)
    private String size;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "logo_url", length = 500)
    private String logoUrl;

    @Column(length = 100)
    private String country;

    @Column(length = 100)
    private String city;

    @Column(nullable = false)
    private Boolean active = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = OffsetDateTime.now();
        updatedAt = OffsetDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = OffsetDateTime.now();
    }
}

File: frontend/src/types/company.ts
export interface CompanyProfile {
  id: number;
  name: string;
  cnpj: string | null;
  website: string | null;
  sector: string | null;
  size: string | null;
  description: string | null;
  logoUrl: string | null;
  country: string | null;
  city: string | null;
  active: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CompanyRegistrationRequest {
  companyName: string;
  sector?: string;
  size?: string;
  website?: string;
  country?: string;
  city?: string;
  description?: string;
  adminName: string;
  email: string;
  password: string;
}

export interface CompanyRegistrationResponse {
  companyId: number;
  token: string;
  message: string;
}

export interface CompanyUpdateRequest {
  name: string;
  cnpj?: string;
  website?: string;
  sector?: string;
  size?: string;
  description?: string;
  logoUrl?: string;
  country?: string;
  city?: string;
}

export interface CompanyDashboard {
  companyName: string;
  activeJobs: number;
  totalCandidates: number;
  profileViews: number;
}
